pkg/api/latest/latest.go

```diff
@@ -18,125 +18,101 @@ package latest
 
 import (
 	"fmt"
-	"sort"
-	"strings"
 
+	"k8s.io/kubernetes/pkg/api"
 	"k8s.io/kubernetes/pkg/api/meta"
 	"k8s.io/kubernetes/pkg/api/registered"
 	"k8s.io/kubernetes/pkg/api/unversioned"
+	"k8s.io/kubernetes/pkg/api/v1"
 	"k8s.io/kubernetes/pkg/runtime"
+	"k8s.io/kubernetes/pkg/util/sets"
 )
 
-var (
-	allGroups = GroupMetaMap{}
-	// Group is a shortcut to allGroups.Group.
-	Group = allGroups.Group
-	// RegisterGroup is a shortcut to allGroups.RegisterGroup.
-	RegisterGroup = allGroups.RegisterGroup
-	// GroupOrDie is a shortcut to allGroups.GroupOrDie.
-	GroupOrDie = allGroups.GroupOrDie
-	// AllPreferredGroupVersions returns the preferred versions of all
-	// registered groups in the form of "group1/version1,group2/version2,..."
-	AllPreferredGroupVersions = allGroups.AllPreferredGroupVersions
-	// IsRegistered is a shortcut to allGroups.IsRegistered.
-	IsRegistered = allGroups.IsRegistered
-)
+// ExternalVersions lists all known external versions for this group from most preferred to least preferred
+var ExternalVersions = []unversioned.GroupVersion{v1.SchemeGroupVersion}
```

Comment:
> ok... looks like what is actually allowed is the intersection of this list's initial value, and what is in registered.supportedVersions (or what is in $KUBE_API_VERSIONS, if specified)... which is computed in init(), ordered by whatever is specified in the registered package, and reassigned here.
> 
> It is pretty misleading to have an initial list and priority order that gets filtered and reordered in init(). I'd rather see something like:

I can't think of a good reason to restrict decoding at all and that's what this does.  If you want to restrict available types, restrict resources.  If you want the client to behave in a particular way, have the client maintain its preferences.  I'm hard pressed to think of any reason why I'd ever want to have the actual list of external versions reordered or some removed entirely.

I did the minimal amount of work to maintain congruence with old functionality until I chase down what we really need and change it.